data_describe/anomaly_detection/detection.py

```diff
@@ -0,0 +1,548 @@
+    if not _is_dataframe(data):
+        raise ValueError("Data frame required")
+
+    # checks if estimator exists and if it follows the .predict and .fit methods
+    if estimator:
+        if not hasattr(estimator, "predict") and not hasattr(estimator, "fit"):
+            raise AttributeError(
+                "Input model does not contain the 'predict' or 'fit' method."
+            )
+    # TODO(truongc2): Update available methods
+    # ml_methods = {
+    #     "timeseries": ["arima"],
+    # }
+
+    numeric_data = data.select_dtypes("number")
+
+    # ensures date_col is a datetime object and sets as datetimeindex
+    if date_col:
+        if date_col != "index":
+            numeric_data.index = pd.to_datetime(data[date_col])
+            if not numeric_data.index.is_monotonic_increasing:
+                numeric_data.sort_index(inplace=True)
+
+    anomalywidget = _get_compute_backend(compute_backend, numeric_data).compute_anomaly(
+        data=numeric_data,
+        target=target,
+        date_col=date_col,
+        estimator=estimator,
+        n_periods=n_periods,
+        data_split=data_split,
+        method=method,
+        **kwargs,
+    )
+
+    anomalywidget.viz_backend = viz_backend
+    anomalywidget.date_col = date_col
+    anomalywidget.target = target
+    anomalywidget.viz_backend = viz_backend
+    anomalywidget.n_periods = n_periods
+    anomalywidget.sigma = sigma
+    anomalywidget.xlabel = date_col
+    anomalywidget.ylabel = target
+
+    return anomalywidget
+
+
+def _pandas_compute_anomaly(
+    data,
+    target: Optional[str] = None,
+    date_col: Optional[str] = None,
+    method: Optional[str] = "arima",
+    estimator=None,
+    n_periods: Optional[int] = None,
+    data_split: Optional[int] = None,
+    **kwargs,
+):
+    """Backend implementation of anomaly detection.
+
+    Args:
+        data (DataFrame): The dataframe.
+        target (str, optional): The target column. Defaults to None. If target is None, unsupervised methods are used.
+        date_col (str, optional): Datetime column if data is timeseries. Defaults to None. If data is timeseries, date_col must be specified.
+        method (str, optional): Select method from this list. Only "arima" is supported.
+        estimator (optional): Fitted or instantiated estimator with a .predict() and .fit() method. Defaults to None.
+            If estimator is instantiated but not fitted, data_split must be specified.
+        n_periods (int, optional): Number of periods for timeseries anomaly detection. Default is None.
+        data_split (int, optional): Index to split the data into a train set and a test set. Defaults to None.
+            data_split must be specified if estimator is instantiated but not fitted.
+        **kwargs: Keyword arguments.
+
+    Raises:
+        ValueError: If method is not implemented.
+
+    Returns:
+        AnomalyDetectionWidget
+
+    """
+    # Timeseries indicator
+    if date_col:
+
```

Comment:
To reduce nesting and make code more readable, use the following pattern instead:

```
if not date_col:
    raise ValueError(...)

if not target:
    raise ValueError(...)

...

predictions_df = stepwise_fit_predict(...)
```

This accomplishes the same while keeping the error close to the if statement and prevents multiple levels of if-block nesting

Reply:
@haishiro  date_col and target col arguments are optional arguments so it shouldn't raise a ValueError